ci/custom-triplets/arm64-osx-dynamic-release.cmake

```diff
@@ -3,7 +3,7 @@ set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
 set(VCPKG_OSX_ARCHITECTURES arm64)
 set(VCPKG_BUILD_TYPE release)
-set(VCPKG_OSX_DEPLOYMENT_TARGET "12.0")
+set(VCPKG_OSX_DEPLOYMENT_TARGET "14.0")
```

Comment:
I think we want this to be our minimum supported version of MacOS, even if we're building on a newer MacOS runner.  So this should probably be 13 since 12 is now past [EOL](https://endoflife.date/macos).

Reply:
It seems like `MACOSX_DEPLOYMENT_TARGET` is getting ignored during the build because it looks like we are building wheels for only >= 14 (per the errors from when repairing the wheel).  I don't recall this being an issue before, so it might be that we're not setting this when using the `lukka/run-vcpkg@v11` action?

Reply:
Oops, I read that wrong.  This is setting 14 as the minimum version we build for, but we were still expecting `MACOSX_DEPLOYMENT_TARGET` to be 12 in `pyproject.toml`.